core/controllers/reader.py

```diff
@@ -300,7 +300,127 @@ def get(self, exploration_id):
+        super(StorePlaythroughHandler, self).__init__(request, response)
+        self.exp_issues = None
+        self.issue_schema_version = None
+
+    def _is_issue_in_exp_issues_model(
+            self, exp_issues, issue_type, customization_args):
+        """Checks whether the given issue exists in the unresolved issues list
+        of the exploration issues model.
+
+        Args:
+            exp_issues: ExplorationIssues. The exploration issues domain object.
+            issue_type: str. The type of the issue obtained from the playthrough
+                data.
+            customization_args: dict(str: *). The customization args dict
+                obtained from the playthrough data.
+
+        Returns:
+            int|None. The index at which the issue was found, None otherwise.
+        """
+        for index, issue in enumerate(exp_issues.unresolved_issues):
+            if issue.issue_type == issue_type:
+                issue_customization_args = issue.issue_customization_args
+                # In case issue_keyname is 'state_names', the ordering of the
+                # list is important i.e. [a,b,c] is different from [b,c,a].
+                issue_keyname = stats_models.ISSUE_TYPE_KEYNAME_MAPPING[
+                    issue.issue_type]
+                if (issue_customization_args[issue_keyname] ==
+                        customization_args[issue_keyname]):
+                    return index
+        return None
+
+    def _move_playthrough_to_correct_issue(self, playthrough, orig_playthrough):
+        """Moves the updated playthrough to its correct issue in the unresolved
+        issues list.
+
+        Args:
+            playthrough: Playthrough. The updated playthrough domain object.
+            orig_playthrough: Playthrough. The original playthrough domain
+                object.
+        """
+        playthrough_stored = True
+        issue_index = self._is_issue_in_exp_issues_model(
+            self.exp_issues, playthrough.issue_type,
+            playthrough.issue_customization_args)
+        # Check whether the playthrough can be added to its new issue,
+        # if not, it stays in its old issue.
+        if issue_index is not None:
+            if (len(self.exp_issues.unresolved_issues[
+                    issue_index].playthrough_ids) < (
+                        feconf.MAX_PLAYTHROUGHS_FOR_ISSUE)):
+                self.exp_issues.unresolved_issues[
+                    issue_index].playthrough_ids.append(playthrough.id)
+            else:
+                playthrough_stored = False
+        else:
+            issue = stats_domain.ExplorationIssue(
+                playthrough.issue_type,
+                playthrough.issue_customization_args,
+                [playthrough.id], self.issue_schema_version, is_valid=True)
+            self.exp_issues.unresolved_issues.append(issue)
+
+        # Now, remove the playthrough from its old issue.
+        if playthrough_stored:
```

Comment:
nit: rename to `has_old_playthrough`

Reply:
@brianrodri Nope, thats not what this flag is about. There are a few cases when a playthrough wont be stored if the maximum number  of playthroughs per issue has been reached. In this case, playthrough_stored will be set to False.

Reply:
Gotcha, but still, this name seems like an indirect way of describing what you just said. It's more useful to know *why* this block needs to run (so it should help describe the goal of this function), but it currently describes *when* the block will run (which should just be an implementation detail).

Reply:
Okay. Made the name different now. PTAL.